core/tauri/src/manager.rs

```diff
@@ -561,7 +561,14 @@ impl<R: Runtime> WindowManager<R> {
           true,
           // ignore "index.html" just to simplify the url
           if path.to_str() != Some("index.html") {
-            format!("{}/{}", url, path.to_string_lossy())
+            // If the base URL has a trailing slash, we don't need to put an additional slash
+            // between the base URL and the path.
+            let separator = match url.ends_with("/") {
```

Comment:
What if `path` doesn't have a leading slash? I think it is better to check for leading slash in `path` instead.
```suggestion
            let separator = match path.starts_with("/") {
```

Reply:
In that case I reckon we should check both? Because if either one has a slash, no additional slash should be added in between. Should we also remove one of the slashes if the domain has a trailing slash *and* the path has a leading slash?

Checking the path as well makes this a little more complicated and at that point it might be more advisable to use a proper "url join & normalise" function that handles all sorts of edge cases, rather than rolling our own thing with `format!()` in here.

Reply:
yeah that's a valid concern but for the past couple of months we have been adding a slash to `url` no matter what and it worked fine so I don't think there will ever be a trailing slash in `url` but `path` can have a leading slash though. what do you think @lucasfernog ?

Reply:
also if you want to make a separate function to handle both cases, go ahead it will be better.

Reply:
> we have been adding a slash to `url` no matter what and it worked fine so I don't think there will ever be a trailing slash in `url`

I gotta say I'm a little confused by this statement. This PR is a fix for #2281 which is specifically about a trailing slash in `url` that causes it to *not* work fine.

In any case, I'll look into proper URL concatenation and normalisation here because that seems like the better solution anyways. I saw that Tauri already uses the [url](https://crates.io/crates/url) crate internally for the base URLs so I'm gonna explore the codebase a little more to see if we can't just use the [`Url.join()`](https://docs.rs/url/2.2.2/url/struct.Url.html#method.join) function which [does exactly what we need](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=95f4444200e4bfb8e514f931c759c698).

Gonna mark the PR as a draft for now.

Reply:
> > we have been adding a slash to `url` no matter what and it worked fine so I don't think there will ever be a trailing slash in `url`
> 
> I gotta say I'm a little confused by this statement. This PR is a fix for #2281 which is specifically about a trailing slash in `url` that causes it to *not* work fine.

Sorry if I wasn't clear, the trailing slash is manually added by us for quite some time now, so you can assume that `url` will never have a trailing slash.

But yeah it is better to check for trailing and leading slashes